Sources/searchable-collection-vc/SearchableCollectionViewController.swift

```diff
@@ -0,0 +1,52 @@
+//
+//  SearchableCollectionViewController.swift
+//  Demo
+//
+//  Created by Yuliia Veresklia on 20/04/2017.
+//
+//
+
+import Foundation
+import UIKit
+
+class SearchableCollectionController: UIViewController {
+    
+    lazy var collectionView: UICollectionView = {
+        let layout = SearchableCollectionViewLayout()
+        
+        let view = UICollectionView(frame: CGRect.zero, collectionViewLayout: layout)
+        view.translatesAutoresizingMaskIntoConstraints = false
+        (view as UIScrollView).delegate = self
```

Comment:
You don't need to cast here because UICollectionView is a subclass of UIScrollView.